rustls/src/record_layer.rs

```diff
@@ -246,6 +241,18 @@ pub(crate) struct Decrypted<'a> {
     pub(crate) plaintext: InboundPlainMessage<'a>,
 }
 
+#[derive(Debug, Eq, PartialEq)]
+pub(crate) enum PreEncryptAction {
+    /// No action is needed before calling `encrypt_outgoing`
+    Nothing,
+
+    /// A `close_notify` alert should be sent ASAP
+    Close,
+
+    /// Do not call `encrypt_outgoing` further, it will panic.
```

Comment:
nit: maybe worth mentioning why it would panic? e.g. something about too many messages having been encrypted.